knowledge_graph/network_class.py

```diff
@@ -107,28 +91,29 @@ def dfs_for_classes(self, node):
         if classes:
 
             for ont_class in classes:
-                if ont_class != owl.Thing:
+                # if ont_class != owl.Thing:
+                if isinstance(ont_class, owlready2.entity.ThingClass):
```

Comment:
No super sure about this. This dfs is the most complicated aspect of the project. Would need @andrew-banister to review this aspect.

Reply:
Fair. That's partly why I included the regression test: this bit is complicated, and I wanted a concrete way to demonstrate that it's at least generating the same graph.

Reply:
The way the current ontology is, not every edge case is tested. @andrew-banister  mentioned this to me when we wrote this double depth first search aspect of the code.

Reply:
Looks good. I was confused because I thought isinstance was an Owlready2 function and instance was referring to ontology instance. But it’s not it’s referring just to standard python object type, which makes sense now.